datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -2162,4 +2194,56 @@ mod tests {
         assert_eq!(res, common_requirement);
         Ok(())
     }
+
+    #[test]
+    fn test_agg_exec_same_schema() -> Result<()> {
```

Comment:
is there any way that this can be tested from SQL? Or does it require a programatic test?

Reply:
Actually I tried to produce this error from a SQL test. However couldn't reproduce it. The reason may be related to the some rules in physical optimizer has `schema_check` flag false. But I am not sure, I couldn't come up with a reproducer from the SQL query.